src/service/angular-sails.js

```diff
@@ -1,74 +1,168 @@
-
-                promise.success = function (fn) {
-                    promise.then(fn);
-                    return promise;
-                };
-
-                promise.error = function (fn) {
-                    promise.then(null, fn);
-                    return promise;
-                };
-
-                return deferred;
-            },
-            resolveOrReject = this.responseHandler || function (deferred, data) {
-                // Make sure what is passed is an object that has a status that is a number and if that status is no 2xx, reject.
-                if (data && angular.isObject(data) && data.status && !isNaN(data.status) && Math.floor(data.status / 100) !== 2) {
-                    deferred.reject(data);
-                } else {
-                    deferred.resolve(data);
-                }
-            },
-            angularify = function (cb, data) {
-                $timeout(function () {
-                    cb(data);
-                });
-            },
-            promisify = function (methodName) {
-                socket['legacy_' + methodName] = socket[methodName];
-                socket[methodName] = function (url, data, cb) {
-                    var deferred = defer();
-                    if (cb === undefined && angular.isFunction(data)) {
-                        cb = data;
-                        data = null;
-                    }
-                    deferred.promise.then(cb);
-                    socket['legacy_' + methodName](url, data, function (result) {
-                        resolveOrReject(deferred, result);
-                    });
-                    return deferred.promise;
-                };
-            },
-            wrapEvent = function (eventName) {
-                socket['legacy_' + eventName] = socket[eventName];
-                socket[eventName] = function (event, cb) {
-                    if (cb !== null && angular.isFunction(cb)) {
-                        socket['legacy_' + eventName](event, function (result) {
-                            angularify(cb, result);
-                        });
-                    }
-                };
-            };
-
-        angular.forEach(httpVerbs, promisify);
-        angular.forEach(eventNames, wrapEvent);
-
-        return socket;
+    this.base = '';
+    this.config = {};
+    this.debug = false;
+
+    // like https://docs.angularjs.org/api/ng/service/$http#interceptors
+    // but with sails.io arguments
+    var interceptorFactories = this.interceptors = [
+      /*function($injectables) {
+         return {
+           request: function(config) {},
+           response: function(response) {},
+           requestError: function(config) {},
+           responseError: function(response) {}
+         };
+       }*/
+    ];
+
+    this.$get = ['$q', '$injector', '$rootScope', '$log', function($q, $injector, $rootScope, $log) {
+      var socket = (io.sails || io).connect(provider.url, provider.config);
```

Comment:
So, it turns out that on a certain version of sails.io.js `io.sails` is an object, but it doesn't contain `connect()`.  Something else needs to be done here to make sure `io.sails.connect` exists else use `io.connect`  a one lliner would be `var socket = (io.sails && io.sails.connect || io.connect)(provider.url, provider.config);`